stdlib/array.gr

```diff
@@ -92,6 +92,33 @@ export let mapi = (fn, array) => {
   })
 }
 
+export let fill = (value, array) => {
+  let length = length(array)
+  forEachi((_, index) => {
+    array[index] := value
+  }, array)
+  void
+}
+
+export let fillRange = (value, start, stop, array) => {
+  let length = length(array)
+  let startIndex = if (start < 0) length + start else start
+  let stopIndex = if (stop < 0) length + stop else stop
+  
+  if (startIndex > length) {
+    ignore(fail 'The start index is outside the array')
+  }
+  if (startIndex > stopIndex) {
+    ignore(fail 'The start index cannot be higher than the stop index')
+  }
+  forEachi((_, index) => {
+    if (index >= startIndex && index < stopIndex) {
+      ignore(array[index] := value)
+    }
+  }, array)
+  void
```

Comment:
Here we can use a while loop so we don't have to iterate the whole thing 👌🏽
```suggestion
  let mut index = startIndex
  while (index < stopIndex) {
    array[index] := value
    index += 1
  }
```

Reply:
Yup! Done, just added an extra check for index > length